Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -397,111 +306,218 @@ def _splitBaseAndMarkPairs(self, pairs, marks):
             basePairs[:] = pairs
         return basePairs, markPairs
 
-    def _makeSplitDirectionKernLookups(
-        self, lookups, pairs, ignoreMarks=True, suffix=""
-    ):
-        dfltKern = self._makeKerningLookup(
-            "kern_dflt" + suffix,
-            pairs,
-            exclude=(lambda pair: {"LTR", "RTL"}.intersection(pair.directions)),
-            rtl=False,
-            ignoreMarks=ignoreMarks,
-        )
-        if dfltKern:
-            lookups.setdefault("DFLT", []).append(dfltKern)
-
-        ltrKern = self._makeKerningLookup(
-            "kern_ltr" + suffix,
-            pairs,
-            exclude=(lambda pair: not pair.directions or "RTL" in pair.directions),
-            rtl=False,
-            ignoreMarks=ignoreMarks,
-        )
-        if ltrKern:
-            lookups.setdefault("LTR", []).append(ltrKern)
-
-        rtlKern = self._makeKerningLookup(
-            "kern_rtl" + suffix,
-            pairs,
-            exclude=(lambda pair: not pair.directions or "LTR" in pair.directions),
-            rtl=True,
-            ignoreMarks=ignoreMarks,
-        )
-        if rtlKern:
-            lookups.setdefault("RTL", []).append(rtlKern)
+    def _makeSplitScriptKernLookups(self, lookups, pairs, ignoreMarks=True, suffix=""):
+        bidiGlyphs = self.context.bidiGlyphs
+        glyphScripts = self.context.glyphScripts
+        kerningPerScript = splitKerning(pairs, glyphScripts)
+        for script, pairs in kerningPerScript.items():
+            scriptLookups = lookups.setdefault(script, {})
+
+            key = f"kern_{script}{suffix}"
+            lookup = scriptLookups.get(key)
+            if not lookup:
+                # For neatness:
+                lookup = self._makeKerningLookup(
+                    key.replace(COMMON_SCRIPT, "Common"),  # For neatness
```

Comment:
there's already a `# For neatness:` comment immediately before
```suggestion
                    key.replace(COMMON_SCRIPT, "Common"),
```